addons/account_peppol/wizard/peppol_config_wizard.py

```diff
@@ -0,0 +1,158 @@
+    service_ids = fields.One2many(
+        comodel_name='account_peppol.service',
+        inverse_name='wizard_id',
+        compute='_compute_service_ids',
+        store=True,
+        readonly=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # COMPUTES
+    # -------------------------------------------------------------------------
+
+    @api.depends('account_peppol_proxy_state')
+    def _compute_service_json(self):
+        for wizard in self:
+            if wizard.account_peppol_proxy_state == 'receiver':
+                wizard.service_json = wizard.account_peppol_edi_user._peppol_get_services().get('services')
+            else:
+                wizard.service_json = False
+
+    @api.depends('account_peppol_proxy_state')
+    def _compute_service_info(self):
+        for wizard in self:
+            message = ''
+            if wizard.account_peppol_proxy_state == 'receiver':
+                supported_doctypes = self.env['res.company']._peppol_supported_document_types()
+                if (non_configurable := [
+                    identifier
+                    for identifier in (wizard.service_json or {})
+                    if identifier not in supported_doctypes
+                ]):
+                    message = Markup('%s<ul>%s</ul>') % (
+                        _(
+                            "The following services are listed on your participant but cannot be configured here. "
+                            "If you wish to configure them differently, please contact support."
+                        ),
+                        Markup().join(
+                            Markup('<li>%s</li>') % (wizard.service_json[identifier]['document_name'])
+                            for identifier in non_configurable
+                        ),
+                    )
+            wizard.service_info = message
+
+    @api.depends('account_peppol_proxy_state')
+    def _compute_service_ids(self):
+        """Get the selectable document types.
+
+        Synthesize a combination of locally available document types and those added to the user on
+        the IAP, add the relevant services.
+        """
+        supported_doctypes = self.env['res.company']._peppol_supported_document_types()
+        for wizard in self:
+            if wizard.account_peppol_proxy_state == 'receiver':
+                wizard.service_ids = self.env['account_peppol.service'].create([
+                    {
+                        'document_identifier': identifier,
+                        'document_name': document_name,
+                        'enabled': identifier in (wizard.service_json or {}),
+                        'wizard_id': wizard.id,
+                    }
+                    for identifier, document_name in supported_doctypes.items()
+                ])
+            else:
+                wizard.service_ids = None
+
+    def button_save(self):
```

Comment:
Do you have a better name for this method? One having more meaning if possible :thinking:

Reply:
huuuh `button_sync_form_with_peppol_proxy` could work ? :thinking: